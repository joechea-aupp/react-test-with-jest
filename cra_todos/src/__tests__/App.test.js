import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act, Simulate } from "react-dom/test-utils";
import App from "../App";

let div = null;

beforeEach(() => {
  div = document.createElement("div");
  document.body.appendChild(div);
});

afterEach(() => {
  unmountComponentAtNode(div);
  div.remove();
  div = null;
});

jest.mock("../services", () => {
  return {
    getToDos: () =>
      Promise.resolve([
        {
          title: "Book concert tickets",
          done: false,
          id: 1
        },
        {
          title: "Buy a new iPhone",
          done: true,
          id: 2
        },
        {
          title: "Write an article on React",
          done: true,
          id: 3
        }
      ]),
    addTodo: obj => Promise.resolve({ ...obj, id: 4 })
  };
});

describe("Testing the App", () => {
  it("Renders a snapshot successfully", async () => {
    await act(async () => {
      render(<App />, div);
    });

    expect(div.innerHTML).toMatchSnapshot();
  });

  it("Add task function works", async () => {
    await act(async () => {
      render(<App />, div);
    });

    const inputFld = document.querySelector(".add-todo > input");

    await act(async () => {
      await Simulate.change(inputFld, {
        target: { value: "This is a test task" }
      });
      await Simulate.keyUp(inputFld, { key: "Enter", keyCode: 13 });
    });

    expect(div.innerHTML).toMatchSnapshot();
  });
});
