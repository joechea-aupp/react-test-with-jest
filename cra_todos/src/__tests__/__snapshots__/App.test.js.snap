// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the App Add task function works 1`] = `"<div class=\\"app\\"><div class=\\"title\\">Tasks</div><div class=\\"add-todo\\"><input type=\\"text\\" placeholder=\\"Add a task\\" value=\\"\\"></div><div class=\\"list\\"><div class=\\"list-item\\"><div class=\\"task\\">Book concert tickets</div><div class=\\"status-btn\\"></div><div class=\\"delete-btn\\">X</div></div><div class=\\"list-item\\"><div class=\\"task\\">Buy a new iPhone</div><div class=\\"status-btn status-done\\"></div><div class=\\"delete-btn\\">X</div></div><div class=\\"list-item\\"><div class=\\"task\\">Write an article on React</div><div class=\\"status-btn status-done\\"></div><div class=\\"delete-btn\\">X</div></div><div class=\\"list-item\\"><div class=\\"task\\">This is a test task</div><div class=\\"status-btn\\"></div><div class=\\"delete-btn\\">X</div></div></div></div>"`;

exports[`Testing the App Renders a snapshot successfully 1`] = `"<div class=\\"app\\"><div class=\\"title\\">Tasks</div><div class=\\"add-todo\\"><input type=\\"text\\" placeholder=\\"Add a task\\" value=\\"\\"></div><div class=\\"list\\"><div class=\\"list-item\\"><div class=\\"task\\">Book concert tickets</div><div class=\\"status-btn\\"></div><div class=\\"delete-btn\\">X</div></div><div class=\\"list-item\\"><div class=\\"task\\">Buy a new iPhone</div><div class=\\"status-btn status-done\\"></div><div class=\\"delete-btn\\">X</div></div><div class=\\"list-item\\"><div class=\\"task\\">Write an article on React</div><div class=\\"status-btn status-done\\"></div><div class=\\"delete-btn\\">X</div></div></div></div>"`;
